### PR TITLE
Flexible tracking types

### PR DESCRIPTION
**Pull request by wleese**
Allow all contact modes to be used for Incident Tracking.

https://github.com/linkedin/iris/issues/353

Still a bit rough around the edges, with lot's of the email specific stuff not to mention UI.
Posting early for feedback.

### CODE REVIEW

**Comment by wleese** [comment]
Slowly progressing. Javascript painful.

**Comment by wleese** [comment]
Ready for review.

File: src/iris/bin/sender.py [hunk 1]
```diff
@@ -447,7 +447,20 @@ def escalate():
...
+            else:
+                tracking_message = {
+                    'noreply': True,
+                    'destination': tracking_key,
+                    'mode': tracking_type
+                }
+                try:
+                    body = app_tracking_template['body'].render(**context)
+                except Exception as e:
+                    body = 'plan %s - tracking notification body failed to render: %s' % (plan['name'], str(e))
```
Comment (jrgp) [inline src/iris/bin/sender.py]: You can remove the explicit str() as %s will do that for you

File: src/iris/ui/templates/plan.html [hunk 2]
```diff
@@ -255,6 +258,13 @@ <h4><i class="icon-plus icon">+</i> Add application tracking template</h4>
           <div class="notification-header">HTML</div>
           <pre class="notification-body">{{breakLines email_html}}</pre>
         </div>
+        {{/isEqual}}
+        {{^isEqual ../tracking_type 'email'}}
```
Comment (jrgp) [inline src/iris/ui/templates/plan.html]: I think this should be a # not ^ ?

Comment (wleese) [reply src/iris/ui/templates/plan.html]: While I cannot point to documentation - I couldn't find any. I found a comment on stackoverflow (?) mentioning this usage negates the helpers behavior. I've tested this and it works as intended as far as I can tell.
Also see: https://github.com/linkedin/oncall/pull/128#discussion_r160308027

**Comment by wleese** [comment]
@dwang159 cannot reproduce.

- Go to Plans
- Create New Plan
- Expand Incident Tracking
(At that point, the following settings are prefilled: Type: "call" and  Target: (grey)example@example.com)
- Type in a target
- Click "Add Application Tracking Template"
- Pick an application, fill out the template and Save.
- Go back to Plans tab, open your new plan. .. all tracking settings should be visible (likewise right after the save when the plan is displayed in read mode).

**Comment by dwang159** [comment]
My fault, my database was incorrectly configured. Testing again, seems to work fine.

**Pull request merged**